### PR TITLE
Skr tests

### PR DESCRIPTION
**Pull request by MarekMichali**
**Description**

Add skr tests from kyma repository

Changes proposed in this pull request:

**Related issue(s)**

#3

### CODE REVIEW

**Review by PK85 — DISMISSED** [review]

**Review by PK85 — COMMENTED** [review]
rename testing/fast-integration to tests/e2e

**Review by PK85 — APPROVED** [review]

File: testing/e2e/skr/skr-test/README.md [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+### With provisioning
+
+In this mode, the test executes the following steps:
+
+1. Provision an SKR cluster.
+2. Run the OIDC test.
+3. Deprovision the SKR instance and clean up the resources.
+
+### Without Provisioning.
```
Comment (IwonaLanger) [inline testing/e2e/skr/skr-test/README.md:21]:
```suggestion
### Without provisioning
```

File: testing/e2e/skr/skr-test/README.md [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+In this mode the test additionally needs the following environment variables:
+- `SKIP_PROVISIONING`, set to `true`
+- `INSTANCE_ID` the uuid of the provisioned SKR instance
```
Comment (IwonaLanger) [inline testing/e2e/skr/skr-test/README.md:25]:
```suggestion
- **SKIP_PROVISIONING** set to `true`
- **INSTANCE_ID** the UUID of the provisioned SKR instance
```

File: testing/e2e/skr/skr-test/README.md [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+
+In this mode, the test executes the following steps:
+1. Ensure the SKR exists.
+2. Run the OIDC test.
+3. Clean up the resources.
```
Comment (IwonaLanger) [inline testing/e2e/skr/skr-test/README.md:30]:
```suggestion
1. Ensures the SKR exists.
2. Runs the OIDC test.
3. Cleans up the resources.
```

File: testing/e2e/skr/skr-test/README.md [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+
+You can use the SKR test in two modes - with and without provisioning.
+
+### With provisioning
+
+In this mode, the test executes the following steps:
+
+1. Provision an SKR cluster.
+2. Run the OIDC test.
+3. Deprovision the SKR instance and clean up the resources.
```
Comment (IwonaLanger) [inline testing/e2e/skr/skr-test/README.md:19]:
```suggestion
1. Provisions an SKR cluster.
2. Runs the OIDC test.
3. Deprovisions the SKR instance and clean up the resources.
```

File: testing/e2e/skr/skr-test/README.md [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+In this mode, the test executes the following steps:
+1. Ensure the SKR exists.
+2. Run the OIDC test.
+3. Clean up the resources.
+ 
+>**NOTE:** The SKR test additionally contains a stand-alone script that you can use to register the resources.
+
+## Test execution
+
+1. Before you run the test, prepare the `.env` file based on the following `.env.template`:
```
Comment (IwonaLanger) [inline testing/e2e/skr/skr-test/README.md:36]: If there is another, exactly the same, `.env.template`, you might just link to it.

File: testing/e2e/skr/skr-test/README.md [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+KCP_TECH_USER_PASSWORD=
+KCP_MOTHERSHIP_API_URL=
+KCP_KUBECONFIG_API_URL=
+
+BTP_OPERATOR_CLIENTID=
+BTP_OPERATOR_CLIENTSECRET=
+BTP_OPERATOR_URL=
+BTP_OPERATOR_TOKENURL=
+
+AL_SERVICE_KEY= #must be a cloud foundry service key with the info about `UAA (User Account and Authentication)`. Learn more about [Managing Service Keys in Cloud Foundry](https://docs.cloudfoundry.org/devguide/services/service-keys.html).
```
Comment (IwonaLanger) [inline testing/e2e/skr/skr-test/README.md:65]:
```suggestion
AL_SERVICE_KEY= #must be a Cloud Foundry service key with the info about UAA (User Account and Authentication). To learn more about Managing Service Keys in Cloud Foundry, go to: https://docs.cloudfoundry.org/devguide/services/service-keys.html.
```

File: testing/e2e/skr/skr-test/README.md [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+```
+
+2. To set up the environment variables in your system, run:
+
+```bash
+export $(xargs < .env)
+```
+
+3. Choose whether you want to run the test with or without provisioning.
+   - To run the test **with** provisioning, call the following target:
```
Comment (IwonaLanger) [inline testing/e2e/skr/skr-test/README.md:75]:
```suggestion
   - To run the test with provisioning, call the following target:
```

File: testing/e2e/skr/skr-test/README.md [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+
+    ```bash
+    npm run skr-test
+    #or
+    make skr
+    ```
+    - To run the SKR test **without** provisioning, use the following command:
```
Comment (IwonaLanger) [inline testing/e2e/skr/skr-test/README.md:82]:
```suggestion
    - To run the SKR test without provisioning, use the following command:
```

File: testing/e2e/skr/skr-test/helpers.js [hunk 9]
```diff
@@ -0,0 +1,188 @@
...
+}
+
+function withCustomParams(customParams) {
+  return function(options) {
+    options.customParams = customParams;
+  };
+}
+
+function gatherOptions(...opts) {
+  // If no opts provided the options object will be set to these default values.
```
Comment (IwonaLanger) [inline testing/e2e/skr/skr-test/helpers.js:59]:
```suggestion
  // If no opts provided, the options object will be set to these default values.
```

File: testing/e2e/skr/skr-test/helpers.js [hunk 10]
```diff
@@ -0,0 +1,188 @@
...
+  const options = {
+    instanceID: uuid.v4(),
+    testNS: testNS,
+    // These options are not meant to be rewritten apart from env variable for KEB_USER_ID
+    // If that's needed please add separate function that overrides this field.
```
Comment (IwonaLanger) [inline testing/e2e/skr/skr-test/helpers.js:64]:
```suggestion
    // If that's needed, add a separate function that overrides this field.
```

File: testing/e2e/skr/skr-test/helpers.js [hunk 11]
```diff
@@ -0,0 +1,188 @@
...
+
+  return options;
+}
+
+async function getSKRRuntimeStatus(instanceID) {
+  const runtimeStatus = await kcp.getRuntimeStatusOperations(instanceID);
+  return JSON.parse(runtimeStatus);
+}
+
+// gets the skr config by its instance id
```
Comment (IwonaLanger) [inline testing/e2e/skr/skr-test/helpers.js:106]:
```suggestion
// gets the SKR config by its instance id
```

File: testing/e2e/skr/smctl/helpers.js [hunk 12]
```diff
@@ -0,0 +1,225 @@
...
+    const {stdout} = await execa('smctl', args);
+    if (stdout !== 'Service Binding successfully deleted.') {
+      errors = errors.concat([`failed "smctl ${args.join(' ')}": ${stdout}`]);
+    }
+  } catch (error) {
+    errors = errors.concat([`failed "smctl ${args.join(' ')}": ${error.stderr}\n${error}`]);
+  }
+
+  try {
+    // hint: probably should fail cause that instance created other instannces (after the migration is done)
```
Comment (IwonaLanger) [inline testing/e2e/skr/smctl/helpers.js:149]:
```suggestion
    // hint: probably should fail cause that instance created other instances (after the migration is done)
```

**Review by IwonaLanger — APPROVED** [review]

**Review by PK85 — APPROVED** [review]

**Pull request merged**